### PR TITLE
G20: priority-sorted status rows in sidebar

### PR DESCRIPTION
**Pull request by daveowenatl**
## Summary

- Replace the hardcoded `agent.task` + `agent.message` rendering blocks in the workspace sidebar with a single priority-sorted loop. Other keyed entries (`claude.tool`, `claude.notification`, `claude.subagent`, `gemini.*`, user CLI keys) now surface in the row so all status publishers render, ordered `task (80) > notification (70) > subagent (65) > tool/message (60)`.
- Row color derives from priority — entries ≥ `TASK` tier use the agent-status color, lower tiers use the subtitle grey.
- Dedup for the legacy dual-write pattern: drop any non-`agent.*` entry whose text matches an `agent.*` entry's text so `claude_hook`'s parallel writes don't duplicate rows.
- Extract `build_status_rows()` as a pure function; add three unit tests (priority ordering, dual-write dedup, empty-text filter).

Part of the sidebar parity plan (#260), item **G20**.

## Test plan

- [x] `cargo fmt --check`
- [x] `cargo clippy --workspace -- -D warnings`
- [x] `cargo test -p amux-app` (31 passed; 3 new tests for `build_status_rows`)
- [ ] Smoke-test: run Claude Code in a workspace; `PreToolUse` hook should surface "Reading …" once (not twice) in the sidebar row.
- [ ] Smoke-test: publish via `amux set-status --key user.task --priority 80 --text "Custom task"`; row should render alongside the agent rows in priority order.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/amux-app/src/sidebar.rs [hunk 1]
```diff
@@ -108,6 +108,29 @@ pub(crate) fn paint_close_x(
...
+    status
+        .displayed_by_priority()
+        .into_iter()
+        .filter(|(k, _)| *k != amux_notify::KEY_AGENT_LABEL)
+        .filter(|(_, e)| !e.text.is_empty())
+        .filter(|(k, e)| {
+            k.starts_with(amux_notify::AGENT_KEY_PREFIX) || !agent_texts.contains(e.text.as_str())
+        })
+        .map(|(_, e)| (e.text.clone(), e.priority))
+        .collect()
```
[the remark this answers is not in the document]
Comment (daveowenatl) [reply crates/amux-app/src/sidebar.rs:135]: Fixed. Changed `build_status_rows` to return `Vec<(&str, i32)>` — rows now borrow directly from `WorkspaceStatus.displayed` and allocate nothing per frame. The render loop already consumed `&str` so callers are unchanged.

File: crates/amux-app/src/sidebar.rs [hunk 2]
```diff
@@ -964,4 +969,89 @@ mod tests {
         assert_eq!(badge_label(100), "99+");
         assert_eq!(badge_label(12_345), "99+");
     }
+
+    // Helper: populate the `displayed` snapshot via a real commit pass,
+    // matching what the per-frame sidebar render sees.
+    fn commit(store: &mut amux_notify::NotificationStore) {
+        store.commit_displayed_at(
+            std::time::Instant::now() + std::time::Duration::from_secs(1),
+            std::time::Duration::from_millis(40),
```
[the remark this answers is not in the document]
Comment (daveowenatl) [reply crates/amux-app/src/sidebar.rs]: Fixed. Test helper now uses `NotificationStore::DEBOUNCE_WINDOW` instead of the hardcoded `Duration::from_millis(40)`.

File: crates/amux-app/src/sidebar.rs [hunk 3]
```diff
@@ -108,6 +108,29 @@ pub(crate) fn paint_close_x(
...
 
+/// Build the priority-sorted list of keyed status rows to render for a
+/// workspace. See the call site in `render_workspace_row` for the full
+/// rationale (G20). Returned tuples are `(text, priority)`; priority is
+/// used by the render loop to pick row color.
+fn build_status_rows(status: &amux_notify::WorkspaceStatus) -> Vec<(String, i32)> {
+    let agent_texts: std::collections::HashSet<&str> = status
+        .displayed
+        .iter()
+        .filter(|(k, _)| k.starts_with(amux_notify::AGENT_KEY_PREFIX))
```
[the remark this answers is not in the document]
Comment (daveowenatl) [reply crates/amux-app/src/sidebar.rs]: Fixed. Good catch — the dedup set is now restricted to just `agent.task` and `agent.message` (the two slots that actually get dual-written by `claude_hook`). `agent.label` is excluded so a user entry with text like "Running" or "Needs input" no longer gets silently dropped. Added a regression test (`build_status_rows_does_not_dedup_against_agent_label_text`).
